tests/integration/run_test.go

```diff
@@ -13,22 +13,63 @@ func runTests(t *testing.T, when spec.G, it spec.S) {
 
 	var service testutil.TestService
 
-	it.Before(func() {
-		service.Create(t, "nginx")
-	})
-
 	it.After(func() {
 		service.Remove()
 	})
 
 	when("rio run is called", func() {
-		it("should create a service with default specifications", func() {
-			runningPods := service.GetRunningPods()
+
+		it.Before(func() {
+			service.Create(t, "nginx", "-p", "80/http")
+		})
+
+		it("should create a service with default specifications that can scale up", func() {
 			assert.Equal(t, 1, service.GetAvailableReplicas(), "should have one available replica")
 			assert.Equal(t, 100, service.GetCurrentWeight())
 			assert.Equal(t, "nginx", service.GetImage())
-			assert.Contains(t, runningPods, service.App.Name)
-			assert.Contains(t, runningPods, "2/2")
+			assert.Contains(t, service.Logs(), "linkerd-init")
+			assert.Contains(t, service.Exec("env"), "KUBERNETES_SERVICE_PORT")
+			runningPods := service.GetRunningPods()
+			for _, pod := range runningPods {
+				assert.Contains(t, pod, service.App.Name)
+				assert.Contains(t, pod, "2/2")
+			}
+
+			service.GenerateLoad("80s", 120)
```

Comment:
Is 80 seconds necessary ?

Reply:
Not for the autoscaling in a single test, but for the CPU of the cluster to actually apply it. I'm not sure how to make this one less flaky or take less time. I'm open to all ideas because I don't like it either.

Reply:
Follow up question, if it only takes 5 seconds does the hey command get shut down or does it try to keep running against a removed service ?

Reply:
Maybe you could do a 10 second load and then check, and loop that until it works.